l3kernel/l3tl.dtx

```diff
@@ -2213,6 +2229,20 @@
 % \end{macro}
 % \end{macro}
 %
+% \begin{macro}{\tl_retokenize:n, \tl_retokenize:V}
+%   A thin wrapper from \tn{scantokens}.
+%    \begin{macrocode}
+\cs_gset_protected:Npn \tl_retokenize:n #1
+  {
+    \group_begin:
+      \int_set:Nn \tex_newlinechar:D { 13 }
+    \exp_after:wN \group_end:
+    \tex_scantokens:D \exp_after:wN { \tl_to_str:n {#1} }
```

Comment:
What happened to the `\scantextokens` idea? Should we try to add something that will (most likely) remove the trailing space? If not, the fact that if this is used in horizontal mode it'll have a trailing space in output should be documented!

Reply:
Exactly the same behaviour won't be possible (the grabbing additional arguments for instance). The space-suppression is possible, but only sort of unreliable. Provided the argument doesn't contain a NUL char (or any other arbitrary char), and doesn't change the category code of that character (the latter will be impossible to check, so we have to hope for the best here), you can put `\char_set_catcode_comment:n { 0 }` in the setup code and put a catcode 12 NUL char at the end of the `\tex_scantokens:D` argument.